From c6d00e0f77a0733b0dbe4fd80ede230b2ffa357a Mon Sep 17 00:00:00 2001
From: karim_adli <karimri@protonmail.com>
Date: Fri, 20 Feb 2026 22:21:33 +0300
Subject: [PATCH 18/19] wineboot: Add WINEBOOT_HIDE_DIALOG environment variable

---
 programs/wineboot/wineboot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
index 7c814d1..9240b6c 100644
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -1511,7 +1511,9 @@ static HWND show_wait_window(void)
 {
     HWND hwnd = CreateDialogParamW( GetModuleHandleW(0), MAKEINTRESOURCEW(IDD_WAITDLG), 0,
                                     wait_dlgproc, (LPARAM)prettyprint_configdir() );
-    ShowWindow( hwnd, SW_SHOWNORMAL );
+    WCHAR *hide = _wgetenv( L"WINEBOOT_HIDE_DIALOG" );
+    if (hide == NULL)
+    	ShowWindow( hwnd, SW_SHOWNORMAL );
     return hwnd;
 }
 
-- 
2.53.0

